src/rdf4cpp/writer/BufWriter.hpp

```diff
@@ -366,6 +366,38 @@ struct BufOutputIteratorWriter : BufWriterBase<BufOutputIteratorWriter<OutIter>,
     }
 };
 
+struct BufWriterOutputIterator {
+    BufWriterParts writer;
+    char buff = '\0';
+    bool write_ok = true;
+
+    using difference_type = ptrdiff_t;
+
+    char& operator*() noexcept {
+        if (writer.cursor->size() == 0) [[unlikely]] {
+            writer.flush(writer.buffer, writer.cursor, 1);
+            if (writer.cursor->size() == 0) [[unlikely]] {
+                write_ok = false;
+                assert(false);
+            }
+        }
+        return *writer.cursor->data();
+    }
+
+    BufWriterOutputIterator& operator++() {
+
+        //*writer.cursor->data() = buff;
+        writer.cursor->advance(1);
+        return *this;
+    }
+    BufWriterOutputIterator operator++(int) {
+        assert(false);
```

Comment:
Why is this assert here?

Reply:
sorry, i forgot to remove the entire class, its unused. (leftover from direct write).